plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityVariant.java

```diff
@@ -13,27 +15,74 @@ public class EntityVariant extends EntityProperty<ElementTag> {
     // @name variant
     // @input ElementTag
     // @description
-    // Controls which variant a wolf is.
-    // A list of valid variants can be found at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Wolf.Variant.html>.
+    // Controls which variant a chicken, cow, pig, or wolf is.
+    // A list of valid chicken variants can be found at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Chicken.Variant.html>.
+    // A list of valid cow variants can be found at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Cow.Variant.html>.
+    // A list of valid pig variants can be found at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Pig.Variant.html>.
+    // A list of valid wolf variants can be found at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Wolf.Variant.html>.
     // -->
 
     public static boolean describes(EntityTag entity) {
-        return entity.getBukkitEntity() instanceof Wolf;
+        return entity.getBukkitEntity() instanceof Wolf
+                || (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21) && entity.getBukkitEntity() instanceof Chicken)
+                || (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21) && entity.getBukkitEntity() instanceof Cow)
+                || (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21) && entity.getBukkitEntity() instanceof Pig);
     }
 
     @Override
     public ElementTag getPropertyValue() {
-        return new ElementTag(Utilities.namespacedKeyToString(as(Wolf.class).getVariant().getKey()));
+        if (getEntity() instanceof Wolf wolf) {
+            return new ElementTag(Utilities.namespacedKeyToString(wolf.getVariant().getKeyOrThrow()));
```

Comment:
Did `Wolf.Variant#getKeyOrThrow` exist on 1.20.6?

Reply:
Actually, pretty sure you can't use that either way - Paper doesn't have these Spigot changes